File: Empresa Adoptfest/proyecto java/backend/src/main/java/com/adoptfest/backend/model/CitaDonacion.java
package com.adoptfest.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Table(name = "citas_donaciones")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CitaDonacion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "donacion_especie_id", nullable = false, unique = true)
    private DonacionEspecie donacionEspecie;

    @Column(nullable = false)
    private LocalDate fecha;

    @Column(nullable = false)
    private LocalTime hora;

    @Column(name = "direccion_recoleccion")
    private String direccionRecoleccion;

    @Column(length = 500)
    private String notas;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private EstadoCitaDonacion estado = EstadoCitaDonacion.PROGRAMADA;
}

File: Empresa Adoptfest/proyecto laragon/app/Models/CitaDonacion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class CitaDonacion extends Model
{
    protected $table = 'citas_donaciones';

    protected $fillable = [
        'user_id',
        'donacion_especie_id',
        'fecha',
        'hora',
        'direccion_recoleccion',
        'notas',
        'estado',
    ];

    public function user()
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    public function donacionEspecie()
    {
        return $this->belongsTo(DonacionEspecie::class, 'donacion_especie_id');
    }
}